src/Core/Scheduler/Providers/PointCloudProvider.js

```diff
@@ -0,0 +1,282 @@
+
+    return [positions, icolors];
+}
+
+let nextuuid = 1;
+
+
+function pointsFromArrayBuffer(ab, readPositionAndColor, hasBbox) {
+    // hardcoded position + color
+    const numPoints = (ab.byteLength - (hasBbox ? 6 * 4 : 0)) / (3 * 4 + 4 * 1);
+
+    const [positions, colors] = readPositionAndColor(ab, numPoints);
+
+    let tightbbox;
+    if (hasBbox) {
+        const view = new DataView(ab, 0, 6 * 4);
+        const min = new THREE.Vector3(view.getFloat32(0, true), view.getFloat32(4, true), view.getFloat32(8, true));
+        const max = new THREE.Vector3(view.getFloat32(12, true), view.getFloat32(16, true), view.getFloat32(20, true));
+        tightbbox = new THREE.Box3(min, max);
+    } else {
+        const min = new THREE.Vector3(positions[0], positions[1], positions[2]);
+        const max = min.clone();
+        for (let i = 1; i < numPoints; i++) {
+            const v = { x: positions[3 * i], y: positions[3 * i + 1], z: positions[3 * i + 2] };
+            min.min(v);
+            max.max(v);
+        }
+        tightbbox = new THREE.Box3(min, max);
+    }
+
+    var geometry = new THREE.BufferGeometry();
+    geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
+    geometry.addAttribute('color', new THREE.BufferAttribute(colors, 4, true));
+
+    // generate unique id for picking
+    const ids = new Uint8Array(4 * numPoints);
+    const base_id = nextuuid++;
+    if (numPoints > 0xffff) {
+        // TODO: fixme
+        // throw new Error('unhandled value');
+    }
+    for (let i = 0; i < numPoints; i++) {
+        // todo numpoints > 16bits
+        const v = (base_id << 16) | i;
+        ids[4 * i + 0] = (v & 0xff000000) >> 24;
+        ids[4 * i + 1] = (v & 0x00ff0000) >> 16;
+        ids[4 * i + 2] = (v & 0x0000ff00) >> 8;
+        ids[4 * i + 3] = (v & 0x000000ff) >> 0;
+    }
+
+    geometry.addAttribute('unique_id', new THREE.BufferAttribute(ids, 4, true));
+    const material = new PointsMaterial();
+    const points = new THREE.Points(geometry, material);
+    points.frustumCulled = false;
+    points.matrixAutoUpdate = false;
+    points.baseId = base_id;
+    points.realPointCount = numPoints;
+    points.tightbbox = tightbbox;
+    points.materials = {
+        color: material,
+    };
+
+    return points;
+}
+
+
+function loadBin(layer, url, readPositionAndColor, hasBbox) {
+    return fetch(url, layer.fetchOptions).then(foo => foo.arrayBuffer()).then(ab =>
+         pointsFromArrayBuffer(ab, readPositionAndColor, hasBbox));
+}
+
+
+PointCloudProvider.prototype.preprocessDataLayer = function preprocessDataLayer(layer) {
+    if (!layer.file) {
+        layer.file = 'cloud.js';
+    }
+    layer.fetchOptions = layer.fetchOptions || {};
+
+    Fetcher.json(`${layer.url}/${layer.file}`, layer.fetchOptions).then((cloud) => {
+        layer.metadata = cloud;
+
+        let bbox;
+
+        if (layer.metadata.scale) {
```

Comment:
The point here was to be able to experiment both in itowns and lopocs while having the minimum code impact.

I could change that, but again, all the file format specific code will move in separate files soon.